src/main/webapp/js/profilePage.js

```diff
@@ -1,13 +1,79 @@
 /**
-* Adds the user's username to the input box when the page is loaded
+* Sets up the functionality of the username input box and the submit button.
 */
-async function getUserName() {
+async function loadProfilePage() {
+  let usernameBox = document.getElementById('userName');
+  let timer; // current ms since last keyup
+  let waitTime = 500; // wait 500ms after last keyup before we ask the server if this username is taken
+
   await fetch('/load-authentication-data').then(response => response.json()).then(async (data) => {
     if (!data.loggedIn) {
       window.location.replace(data.loginUrl);
     }
   });  
   fetch('/load-currentuser-data').then(response => response.json()).then(async (data) => {
-    document.getElementById('userName').value = data.username;
+    usernameBox.value = data.username;
   });
+
+  usernameBox.addEventListener("keyup", () => {
+    if(event.keyCode === 13) { // enter button
+      document.getElementById('submit-username-button').click();
+    } else {
+      clearTimeout(timer);
+      timer = setTimeout(displayAvailability, waitTime);
+    }
+  });
+}
+
+async function isTaken(desiredUsername) {
+  const params = new URLSearchParams();
+  params.append('username', desiredUsername);
+
+  let result;
+  await fetch('/username-available-data', {method: 'post', body: params}).then(response => response.json()).then(taken => {
+    result = taken;
+  });
+  return result;
+}
+
+async function getAvailabilityText(desiredUsername) {
+  if(desiredUsername.length == 0) {
+    return 'Username must be at least 1 character'
+  }
+  if(desiredUsername.length > 20) {
```

Comment:
This is okay but any reason why 20?

Reply:
We probably need a cap on the length; there could be problems with displaying the username if it's too long. As for the number 20, it's chosen arbitrarily. Seems like a good cap.

Reply:
I agree, this seems good.